src/api/projects/mutations/createProject/index.js

```diff
@@ -2,8 +2,23 @@
 import ProjectModel from '@models/Project'
 import type { CreateProjectArgs } from '@models/Project'
 
-const resolver = async (_: any, args: CreateProjectArgs) => {
-  const { errors, project } = await ProjectModel.createProject(args)
+type Context = {
+  request: express$Request
+}
+
+const resolver = async (
+  _: any,
+  args: CreateProjectArgs,
+  { request }: Context
+) => {
+  if (request.user === null) {
+    throw new Error('Please log in.')
```

Comment:
I guess we could export an error from `@constants/error.js` to re-use the same in every resolver?

Reply:
I still think that ideally this should be done in one location and thrown if the operation requires to be authenticated 🤔

Reply:
Like chaining a middleware specifically for GraphQL auth behind our global auth middleware ? How would you know which mutations/queries require auth and which don't ?

Reply:
I thought about it and I think we could totally have something like…
```js
const GRAPHQL_OPERATIONS_WITH_AUTH = ['createProject', …]
```

🤔 

And on the middleware, whitelist based on the operation name that I think we can retrieve?